plugins/qcheck-stm/src/ir_of_gospel.ml

```diff
@@ -88,10 +91,49 @@ let sig_item config s =
   | Sig_val (vd, Nonghost) -> Some (val_desc config vd)
   | _ -> None
 
+let state config sigs =
+  let sut_name = Cfg.get_sut_type_name_str config in
+  let is_sut_decl s =
+    let p t = t.td_ts.ts_ident.id_str = sut_name in
+    match s.sig_desc with
+    | Sig_type (_, ts, _) -> (
+        List.filter p ts |> function
+        | [] -> None
+        | [ t ] -> Some t
+        | _ -> assert false)
+    | _ -> None
+  in
+  let open Reserr in
+  let* ty =
+    match List.filter_map is_sut_decl sigs with
+    | [] -> error (No_sut_type sut_name, Location.none)
+    | [ ty ] -> ok ty
+    | _ -> assert false
+  in
+  let* subst =
+    Fun.flip List.assoc_opt
+    <$> (Ortac_core.Ocaml_of_gospel.core_type_of_tysymbol ty.td_ts
```

Comment:
Maybe `Ortac_core` could be opened, at least locally, to shorten this line and similar ones?